src/main/java/hudson/plugins/ec2/EC2Cloud.java

```diff
@@ -156,8 +143,8 @@
     @CheckForNull
     @Deprecated
     private transient Secret secretKey;
-
-    private final EC2PrivateKey privateKey;
+    @CheckForNull
+    private String sshKeysCredentialsId;
```

Comment:
the `privateKey` variable should not be removed it should be marked as transient

Reply:
@res0nance is there any reason to leave it marked as transient if it is not being used at all?
As far as I understand it it will still be passed through the constructor after a plugin update from a version where it was used and will be migrated to a credential. So this would be left as an unused variable.

Reply:
IIRC it wont call the constructor, it might cause deserialization issues if the field no longer exists

Reply:
My observations in debugging showed that the constructor is called here and the migration seemed to also work but should maybe be checked to be removed in future. I rewrote it to similar to the handling of the other deprecated and transient attributes now.